### PR TITLE
Dashboard: improve data visualization

### PR DESCRIPTION
**Pull request by fballarini**
Show backup time in human format and not like crontab (only english).

### CODE REVIEW

File: root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php [hunk 1]
```diff
@@ -48,7 +48,45 @@ private function readBackup()
         $backup['vfs'] = $br['VFSType'] ? $br['VFSType'] : '-';
         $backup['status'] = $br['status'];
         $backup['time'] = $br['BackupTime'];
-
+        
+        $orario = explode(" ",$backup['time']);
```
Comment (gsanchietti) [inline root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php]: Never use Italian inside the code: the whole code must be written in English.

File: root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php [hunk 2]
```diff
@@ -48,7 +48,45 @@ private function readBackup()
...
+      
+        function formatOre($h){
```
Comment (gsanchietti) [inline root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php]: Move functions in the outer scope and set them as private.
It's a matter of code readability.

File: root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php [hunk 3]
```diff
@@ -48,7 +48,45 @@ private function readBackup()
...
+	        if(strlen($h)==1){return "0".$h;}else{return $h;}
```
Comment (gsanchietti) [inline root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php]: Please respect the general indentation rules: we are not using a strict standard but please write it a readable manner.
Take a look at: https://www.php-fig.org/psr/psr-2/

In this case you could also use a ternary operator or much better http://php.net/manual/en/function.sprintf.php

File: root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php [hunk 4]
```diff
@@ -48,7 +48,45 @@ private function readBackup()
...
+        }
+        
+        function formatMin($m){
```
Comment (gsanchietti) [inline root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php]: See comment above.

File: root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php [hunk 5]
```diff
@@ -48,7 +48,45 @@ private function readBackup()
...
+        }
+        
+        if($orario[1] == "*" && $orario[2] == "*" && $orario[3] == "*" && $orario[4] == "*"){
+	        $backup['time'] = "Every hour at minute ".formatMin($orario[0]);
+        }else if($orario[2] == "*" && $orario[3] == "*" && $orario[4] == "*"){
+	        $backup['time'] = 'Every day at '.formatOre($orario[1]).":".formatMin($orario[0]);
+        }else if($orario[2] == "*" && $orario[3] == "*"){
+			$backup['time'] = "Every ".formatDay($orario[4])." at ".formatOre($orario[1]).":".formatMin($orario[0]);
+        }else if($orario[3] == "*" && $orario[4] == "*"){
+	        $backup['time'] = "Every month on ".$orario[2]." at ".formatOre($orario[1]).":".formatMin($orario[0]);
```
Comment (gsanchietti) [inline root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php]: Please to not hard-code English words which needs to be translated.
In this case you need something like:
```
 $backup['time'] = $view->translate("Every month on ").$hour[2] ...
```

File: root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php [hunk 6]
```diff
@@ -48,7 +48,45 @@ private function readBackup()
...
+      
+        function formatOre($h){
+	        if(strlen($h)==1){return "0".$h;}else{return $h;}
+        }
+        
+        function formatMin($m){
+	         if(strlen($m)==1){return "0".$m;}else{return $m;}
+        }
+        
+        function formatDay($d){
```
Comment (gsanchietti) [inline root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php]: You could covert the entire function in a couple of line, something like:
```
$days = array("Sunday", "Monday" ...)
return $days[$d];
```

Also bear in mind that even the days must be translated!

File: root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php (hunk 2 above)
Comment (fballarini) [reply root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php]: If I move functions out of private readBackup() function, they will not work. Also if I rename them in "private" in the same function...

Comment (gsanchietti) [reply root/usr/share/nethesis/NethServer/Module/Dashboard/SystemStatus/Backup.php]: Yes you can access them, you just need to use the right scope.
Example:
```php
private function formatHour($x)
{
    if(strlen($h)==1){return "0".$h;}else{return $h;}
}

private function readBackup()
{
    ...
    $this->formatHour($h)
    ...
}
```

By the way, you can format the date using just one line of code, and replace the entire function like this:
```php
private function readBackup()
{
    ...
   $backup['time'] = $view->translate('Every day at ').sprintf('%02d', $hour[1]) .":".sprintf('%02d', $hour[0]);
   ...
}
```

Also take a look to your `formatOre` and `formatMin`: the two function are identical! ;)

**Comment by gsanchietti** [comment]
Whoa! This is excellent, thank you!

**Pull request merged**